src/Raven.Server/Documents/Indexes/Debugging/IndexDebugExtensions.cs

```diff
@@ -152,45 +150,39 @@ private static IEnumerable<string> IterateKeys(IIterator it, string prefix, long
+            scope.EnsureDispose(self._contextPool.AllocateOperationContext(out TransactionOperationContext indexContext));
 
-                RavenTransaction tx;
-                scope.EnsureDispose(tx = indexContext.OpenReadTransaction());
+            RavenTransaction tx = indexContext.OpenReadTransaction();
+            scope.EnsureDispose(tx);
 
-                var mapPhaseTree = tx.InnerTransaction.ReadTree(MapReduceIndexBase<MapReduceIndexDefinition, IndexField>.MapPhaseTreeName);
-
-                if (mapPhaseTree == null)
-                {
-                    trees = Enumerable.Empty<ReduceTree>();
-                    return scope;
-                }
-
-                var reducePhaseTree = tx.InnerTransaction.ReadTree(MapReduceIndexBase<MapReduceIndexDefinition, IndexField>.ReducePhaseTreeName);
-
-                if (reducePhaseTree == null)
-                {
-                    trees = Enumerable.Empty<ReduceTree>();
-                    return scope;
-                }
+            var mapPhaseTree = tx.InnerTransaction.ReadTree(MapReduceIndexBase<MapReduceIndexDefinition, IndexField>.MapPhaseTreeName);
+            if (mapPhaseTree == null)
+            {
+                trees = Enumerable.Empty<ReduceTree>();
+                return scope;
+            }
 
-                var mapEntries = new List<FixedSizeTree>(docIds.Length);
-                foreach (var docId in docIds)
-                {
-                    FixedSizeTree mapEntriesTree;
-                    scope.EnsureDispose(mapEntriesTree = mapPhaseTree.FixedTreeFor(docId.ToLower(), sizeof(long)));
-                    mapEntries.Add(mapEntriesTree);
-                }
+            var reducePhaseTree = tx.InnerTransaction.ReadTree(MapReduceIndexBase<MapReduceIndexDefinition, IndexField>.ReducePhaseTreeName);
+            if (reducePhaseTree == null)
+            {
+                trees = Enumerable.Empty<ReduceTree>();
+                return scope;
+            }
 
-                FixedSizeTree typePerHash;
-                scope.EnsureDispose(typePerHash = reducePhaseTree.FixedTreeFor(MapReduceIndexBase<MapReduceIndexDefinition, IndexField>.ResultsStoreTypesTreeName, sizeof(byte)));
+            var mapEntries = new List<FixedSizeTree>(docIds.Length);
+            foreach (var docId in docIds)
+            {
+                FixedSizeTree mapEntriesTree = mapPhaseTree.FixedTreeFor(docId.ToLower(), sizeof(long));
+                mapEntries.Add(mapEntriesTree);
+            }
 
-                trees = IterateTrees(self, mapEntries, reducePhaseTree, typePerHash, indexContext, scope);
+            FixedSizeTree typePerHash = reducePhaseTree.FixedTreeFor(MapReduceIndexBase<MapReduceIndexDefinition, IndexField>.ResultsStoreTypesTreeName, sizeof(byte));
+                
+            trees = IterateTrees(self, mapEntries, reducePhaseTree, typePerHash, indexContext, scope);
 
-                return scope.Delay();
-            }
+            return scope;
```

Comment:
We had `return scope.Delay();` now we have `return scope;` - intentional?

Reply:
Can we avoid touching this file completely? I don't think it's required to make any changes here. Touching it will just make a conflicts during the merge from v5.4 if we change anything there

Reply:
Yes, it is intentional. The `.Delay()` was necessary because we were disposing immediately. Now we just return that scope.

Reply:
> Can we avoid touching this file completely? I don't think it's required to make any changes here. Touching it will just make a conflicts during the merge from v5.4 if we change anything there

We can neglect this one. Shouldn't change anything.